test/unit/connection_string.test.ts

```diff
@@ -573,4 +574,51 @@ describe('Connection String', function () {
       expect(client.s.options).to.have.property(flag, null);
     });
   });
+
+  describe('IPv6 host addresses', () => {
+    it('should not allow multiple unbracketed portless localhost IPv6 addresses', () => {
+      // Note there is no "port-full" version of this test, there's no way to distinguish when a port begins without brackets
+      expect(() => new MongoClient('mongodb://::1,::1,::1/test')).to.throw(
+        /invalid connection string/i
+      );
+    });
+
+    it('should not allow multiple unbracketed portless remote IPv6 addresses', () => {
+      expect(
+        () =>
+          new MongoClient(
+            'mongodb://ABCD:f::abcd:abcd:abcd:abcd,ABCD:f::abcd:abcd:abcd:abcd,ABCD:f::abcd:abcd:abcd:abcd/test'
+          )
+      ).to.throw(MongoRuntimeError);
+    });
+
+    it('should allow multiple bracketed portless localhost IPv6 addresses', () => {
+      const client = new MongoClient('mongodb://[::1],[::1],[::1]/test');
+      expect(client.options.hosts).to.deep.equal([
+        { host: '::1', port: 27017, isIPv6: true, socketPath: undefined },
+        { host: '::1', port: 27017, isIPv6: true, socketPath: undefined },
+        { host: '::1', port: 27017, isIPv6: true, socketPath: undefined }
+      ]);
+    });
+
+    it('should allow multiple bracketed portless localhost IPv6 addresses', () => {
```

Comment:
this test case title is the same as the test case above

Reply:
updated: `remote`